addons/adyen_platforms/models/adyen_account.py

```diff
@@ -0,0 +1,711 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+import os
+import requests
+import uuid
+from werkzeug.urls import url_join
+
+from odoo import api, fields, models, _
+from odoo.http import request
+from odoo.exceptions import UserError, ValidationError
+from odoo.tools import date_utils
+
+from odoo.addons.adyen_platforms.util import AdyenProxyAuth
+
+ADYEN_AVAILABLE_COUNTRIES = ['US', 'AT', 'AU', 'BE', 'CA', 'CH', 'CZ', 'DE', 'ES', 'FI', 'FR', 'GB', 'GR', 'HR', 'IE', 'IT', 'LT', 'LU', 'NL', 'PL', 'PT']
+TIMEOUT = 60
+
+
+class AdyenAddressMixin(models.AbstractModel):
+    _name = 'adyen.address.mixin'
+    _description = 'Adyen for Plaforms Address Mixin'
+
+    country_id = fields.Many2one('res.country', string='Country', domain=[('code', 'in', ADYEN_AVAILABLE_COUNTRIES)], required=True)
+    country_code = fields.Char(related='country_id.code')
+    state_id = fields.Many2one('res.country.state', string='State', domain="[('country_id', '=?', country_id)]")
+    state_code = fields.Char(related='state_id.code')
+    city = fields.Char('City', required=True)
+    zip = fields.Char('ZIP', required=True)
+    street = fields.Char('Street', required=True)
+    house_number_or_name = fields.Char('House Number Or Name', required=True)
+
+
+class AdyenIDMixin(models.AbstractModel):
+    _name = 'adyen.id.mixin'
+    _description = 'Adyen for Platforms ID Mixin'
+
+    id_type = fields.Selection(string='Photo ID type', selection=[
+        ('PASSPORT', 'Passport'),
+        ('ID_CARD', 'ID Card'),
+        ('DRIVING_LICENSE', 'Driving License'),
+    ])
+    id_front = fields.Binary('Photo ID Front', help="Allowed formats: jpg, pdf, png. Maximum allowed size: 4MB.")
+    id_front_filename = fields.Char()
+    id_back = fields.Binary('Photo ID Back', help="Allowed formats: jpg, pdf, png. Maximum allowed size: 4MB.")
+    id_back_filename = fields.Char()
+
+    def write(self, vals):
+        res = super(AdyenIDMixin, self).write(vals)
```

Comment:
self could contains more then one record. This is not very likely but the orm allow that, it should take that case into account